src/algorithm/hnswlib/hnswalg.cpp

```diff
@@ -71,8 +71,19 @@ HierarchicalNSW::HierarchicalNSW(SpaceInterface* s,
     rev_size_ = 1.0 / mult_;
 }
 
+void
+HierarchicalNSW::reset() {
+    allocator_->Deallocate(element_levels_);
```

Comment:
Here need clean freed memory pointer (set to nullptr)

Reply:
done